packages/block-library/src/column/edit.native.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withSelect } from '@wordpress/data';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import {
+	InnerBlocks,
+	BlockControls,
+	BlockVerticalAlignmentToolbar,
+} from '@wordpress/block-editor';
+import { Toolbar, ToolbarButton } from '@wordpress/components';
+import { withViewportMatch } from '@wordpress/viewport';
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+
+function ColumnEdit( {
+	attributes,
+	setAttributes,
+	hasChildren,
+	isSelected,
+	getStylesFromColorScheme,
+	isParentSelected,
+	isDescendantOfParentSelected,
+	columnsContainerSettings,
+	isMobile,
+} ) {
+	const { verticalAlignment } = attributes;
+	const {
+		columnsInRow = 1,
+		columnsContainerWidth,
+	} = columnsContainerSettings;
+
+	const columnContainerBaseWidth = styles[ 'column-container-base' ].maxWidth;
+	const containerMaxWidth = styles[ 'columns-container' ].maxWidth;
+
+	const containerWidth = columnsContainerWidth || containerMaxWidth;
+
+	const minWidth = Math.min( containerWidth, columnContainerBaseWidth );
+	const columnBaseWidth = minWidth / columnsInRow;
+
+	const applyBlockStyle = ( placeholder = false ) => {
+		if ( isMobile ) {
+			return;
+		}
+
+		const pullWidth = ( name ) =>
+			( styles[ `column-${ name }-margin` ] || {} ).width || 0;
+
+		let width = columnBaseWidth;
+
+		if ( isParentSelected ) {
+			width -= pullWidth(
+				placeholder ? 'placeholder-selected' : 'parent-selected'
+			);
+		} else if ( isSelected && ! placeholder ) {
+			width -= ! hasChildren
+				? pullWidth( 'selected' )
+				: pullWidth( 'descendant-selected' );
+		} else if ( isDescendantOfParentSelected ) {
+			if ( placeholder ) {
+				width -= pullWidth( 'selected' );
+			} else {
+				width += pullWidth( 'descendant-selected' );
+			}
+		} else if ( placeholder ) {
+			width -=
+				columnsInRow === 1
+					? pullWidth( 'parent-selected' )
+					: pullWidth( 'placeholder-multicol' );
+		}
+
+		return { width };
+	};
```

Comment:
hey @jbinda 👋Actually I find it hard to understand what this code is doing. Could you explain the behavior we want to achieve here? It looks like a pretty complex logic to me. It would really help if you write down the use cases we want in a **non-technical** way.

Reply:
This logic is implemented to compensate the InnerBlock margins that is applied on selected block and its children and the width of border.

`pullWidth()` function takes the proper margin size from .scss style (there is some calculation on variables implemented during navigate-down feature)

The `placeholder` flag means that the Block is empty and it shows placeholder when deselected and AppenderButton when selected 

Then we have a few conditions based on dependencies to block Parent and Selected Block. Depending on that we decrease the width of a Column to fit in available container size to prevent other Column in row to wrap.

Does it make more sense now ?

Reply:
I have also did some refactor on this fragment and it should be more readable. After that here'w how the `applyBlockStyle()` function looks like

```jsx

	const applyBlockStyle = ( placeholder = false ) => {
		if ( isMobile ) {
			return;
		}

		const pullWidths = ( names ) =>
			names.map(
				( name ) =>
					( styles[ `column-${ name }-margin` ] || {} ).width || 0
			);

		let width = columnBaseWidth;
		const names = [
			'selected',
			'parent-selected',
			'descendant-selected',
			'placeholder-selected',
		];
		const [
			selected,
			parentSelected,
			descendantSelected,
			placeholderSelected,
		] = pullWidths( names );

		if ( isParentSelected ) {
			width -= parentSelected;
			return { width };
		}

		if ( placeholder ) {
			if ( isDescendantOfParentSelected ) {
				width -= selected;
			} else {
				width -=
					columnsInRow === 1 ? parentSelected : placeholderSelected;
			}
		} else if ( isSelected ) {
			width -= ! hasChildren ? selected : descendantSelected;
		} else if ( isDescendantOfParentSelected ) {
			width += descendantSelected;
		}

		return { width };
	};

```

If thats look better I will push changes

Reply:
After spending ~1h on this logic unfortunately I am still not understanding all the detailed adjustments done here. 

What you posted in the comment looks better but still uses the same approach. I feel like some simplifications are needed on our approach but since I couldn't understand the need underneath I was not able to suggest something.

Like, for the selected state, when we extract the left/right space I marked with red, doesn't that give us the total width of columns? What I don't understand is I think, why is it not enough to just share the rest of the space equally between columns?

<img width="872" alt="Screen Shot 2020-02-10 at 18 24 48" src="https://user-images.githubusercontent.com/5032900/74164785-4a322780-4c35-11ea-8465-a7753a8a9998.png">

Reply:
To get the `Columns` width I use `onLayout` calculate and store number of Column to show in row and container width in `BlockListSettings` of `Columns` Block.

Then in Column I refer to this values and calculate final width base on them. It split available space between Columns equally.

Then comes the selection border styling that adds additional pixels applying the borders and margins. Thats why I need to compensate it with above.

Hope it make more sense now ?

 I will try to review it once more and try to simplify it